DisCatSharp.ApplicationCommands/ApplicationCommandsExtension.cs

```diff
@@ -2141,4 +2150,189 @@ internal class ContextMenuCommand
+                slashCommands.ForEach(sc => options.Add(new DiscordApplicationCommandAutocompleteChoice(sc.Name, sc.Name.Trim())));
+                return options.AsEnumerable();
+            }
+        }
+
+        public class DefaultHelpAutoCompleteLevelOneProvider : IAutocompleteProvider
+        {
+            public async Task<IEnumerable<DiscordApplicationCommandAutocompleteChoice>> Provider(AutocompleteContext context)
+            {
+                var options = new List<DiscordApplicationCommandAutocompleteChoice>();
+                var globalCommands = await context.Client.GetGlobalApplicationCommandsAsync();
+                var guildCommands = await context.Client.GetGuildApplicationCommandsAsync(context.Guild.Id);
+                var slashCommands = new List<DiscordApplicationCommand>();
+                slashCommands.AddRange(globalCommands);
+                slashCommands.AddRange(guildCommands);
+                var command = slashCommands.FirstOrDefault(ac =>
+                    ac.Name.ToLower().Equals(context.Options[0].Value.ToString().ToLower().Trim()));
+                if (command is null || command.Options is null)
+                {
+                    options.Add(new DiscordApplicationCommandAutocompleteChoice("no_options_for_this_command", "no_options_for_this_command"));
+                }
+                else
+                {
+                    command.Options.ToList().ForEach(option => options.Add(new DiscordApplicationCommandAutocompleteChoice(option.Name, option.Name)));
+                }
+                return options.AsEnumerable();
+            }
+        }
+
+        public class DefaultHelpAutoCompleteLevelTwoProvider : IAutocompleteProvider
+        {
+            public async Task<IEnumerable<DiscordApplicationCommandAutocompleteChoice>> Provider(AutocompleteContext context)
+            {
+                var options = new List<DiscordApplicationCommandAutocompleteChoice>();
+                var globalCommands = await context.Client.GetGlobalApplicationCommandsAsync();
+                var guildCommands = await context.Client.GetGuildApplicationCommandsAsync(context.Guild.Id);
+                var slashCommands = new List<DiscordApplicationCommand>();
+                slashCommands.AddRange(globalCommands);
+                slashCommands.AddRange(guildCommands);
+                var command = slashCommands.FirstOrDefault(ac =>
+                    ac.Name.ToLower().Equals(context.Options[0].Value.ToString().ToLower().Trim()));
+                if (command.Options is null)
+                {
+                    options.Add(new DiscordApplicationCommandAutocompleteChoice("no_options_for_this_command", "no_options_for_this_command"));
+                    return options.AsEnumerable();
+                }
+                var foundCommand = command.Options.ToList().FindAll(op => op.Name.ToLower().Equals(context.Options[1].Value.ToString().ToLower().Trim())).FirstOrDefault();
```

Comment:
same casing comment as above. Additionally, it appears you only want the first or default of this resultset... so move the search criteria from `FindAll` into `FirstOrDefault`